compiler/modules/front/resolveTypes.bal

```diff
@@ -267,7 +268,8 @@ function resolveTypeDesc(ModuleSymbols mod, s:ModuleLevelDefn modDefn, int depth
                 a.push(arg.td);
             }
             t:SemType[] args = from var x in a select check resolveTypeDesc(mod, modDefn, depth + 1, x);
-            t:SemType ret = check resolveTypeDesc(mod, modDefn, depth + 1, td.ret);
+            s:TypeDesc? retTy = td.ret;
+            t:SemType ret = check resolveTypeDesc(mod, modDefn, depth + 1, retTy?: { startPos: td.endPos, endPos: td.endPos, builtinTypeName: "null" });
```

Comment:
If td.ret is (), then the SemType is t:NIL.

Reply:
fixed